naga/src/front/spv/mod.rs

```diff
@@ -3944,7 +3961,68 @@ impl<I: Iterator<Item = u32>> Frontend<I> {
                     );
                     emitter.start(ctx.expressions);
                 }
-                _ => return Err(Error::UnsupportedInstruction(self.state, inst.op)),
+                Op::AtomicIIncrement => {
+                    inst.expect(6)?;
+                    let start = self.data_offset;
+                    let result_type_id = self.next()?;
+                    let result_id = self.next()?;
+                    let pointer_id = self.next()?;
+                    let scope_id = self.next()?;
+                    let memory_semantics_id = self.next()?;
+                    // Store the op for a later pass where we "upgrade" the pointer type
+                    let atomic = AtomicOp {
+                        instruction: AtomicOpInst::AtomicIIncrement,
+                        result_type_id,
+                        result_id,
+                        pointer_id,
+                        scope_id,
+                        memory_semantics_id,
+                    };
+                    self.lookup_atomic.insert(pointer_id, atomic);
+
+                    log::trace!("\t\t\tlooking up expr {:?}", pointer_id);
+                    let (p_lexp_handle, p_lexp_ty_id) = {
+                        let lexp = self.lookup_expression.lookup(pointer_id)?;
+                        (lexp.handle, lexp.type_id)
+                    };
+                    log::trace!("\t\t\tlooking up type {:?}", pointer_id);
+                    let p_ty = self.lookup_type.lookup(p_lexp_ty_id)?;
+
+                    // Create an expression for our result
+                    let r_lexp_handle = {
+                        let expr = crate::Expression::AtomicResult {
+                            ty: p_ty.handle,
+                            comparison: false,
+                        };
```

Comment:
I'm not sure I'm following this exactly, but I *believe* that `pointer_id` is going to be an `OpTypePointer`, so `p_ty.handle` will refer to a Naga `TypeInner::Pointer`. However, an `AtomicResult` expression's type should always be `i32` or `u32` for this instruction - that is, the pointer's base type. I think you can get this by looking up `p_ty.base_id`.